### DISCUSSION TITLE
terminated ec2 without backup

### DISCUSSION THREAD

**Discussion by joshweissmann** [getredash/redash#6135] [Q&A] [answered]
Someone terminated our ec2 instance without a backup. Is there any way to recover the queries stored on redash?

**Comment by justinclift** [comment, marked as answer]
The queries are stored in a PostgreSQL database on the Redash server itself.

If (as it sounds like) your Redash instance has been completely killed... then it's all gone.  :frowning:

It's not super hard to take backups (there are several potential approaches), so it's pretty unfortunate.
